### PR TITLE
feat: add traffic routing policies

### PR DESCRIPTION
**Pull request by hacdias**
License: MIT
Signed-off-by: Henrique Dias <hacdias@gmail.com>

### CODE REVIEW

**Review by nonsense — APPROVED** [review]

File: network/types.go [hunk 1]
```diff
@@ -111,4 +111,8 @@ type Config struct {
 	// A zero value falls back to runenv.TestInstanceCount (i.e. all instances
 	// participating in the test run).
 	CallbackTarget int `json:"-"`
+
+	// WhitelistAll whitelists all traffic through the data plane (except for the
+	// control traffic).
+	WhitelistAll bool
```
Comment (raulk) [inline network/types.go]: Hm, how about we switch this to an enum field `RoutingPolicy`? That would be more future-proof.

```go
type RoutingPolicyType string

const (
    WhitelistAll = RoutingPolicyType("whitelist_all")
)

type Config struct {
    ...
    
    RoutingPolicy RoutingPolicyType
}
```

**Review by raulk — CHANGES_REQUESTED** [review]

File: network/types.go (hunk 1 above)
Comment (nonsense) [reply network/types.go]: Maybe use `AllowList` and `DenyList` (or `BlockList`) - it sounds similar to iptables terminology.

Comment (raulk) [reply network/types.go]: Oh yeah. There terms are so embedded in our ways... Let's avoid controversial terminology. `allow_all` and `deny_all` (future) are the right ways to go here.

File: network/types.go [hunk 2]
```diff
@@ -73,6 +73,15 @@ type LinkRule struct {
 	Subnet net.IPNet
 }
 
+// RoutingPolicyType defines a certain routing policy to a network.
+type RoutingPolicyType string
+
+const (
+	// AllowAll whitelists all traffic through the data plane (except for the
+	// control traffic).
+	AllowAll = RoutingPolicyType("allow_all")
```
Comment (nonsense) [inline network/types.go:80]: In the other PR I see that we are checking for `AllowAll` policy and `default`. I think it'd be nice if we are explicit about the policy, and always have one, so maybe worth adding `DenyAll`. WDYT?

Comment (hacdias) [reply network/types.go:80]: Right, it makes sense. Should we make it a mandatory flag? We'd need to change all the tests. Right now, DenyAll would be the default.

Comment (hacdias) [reply network/types.go:80]: So I just added `DenyAll` and made a few updates to the docs. I will update the corresponding PR on testground repo and see how it looks best.

**Comment by hacdias** [comment]
I'm merging this since it looks we came into an agreement here. We can choose to change later.

**Pull request merged**